python/egobox/egobox.pyi

```diff
@@ -0,0 +1,488 @@
+            y_doe (array[ns, 1 + n_cstr]): ns values of objective and constraints
+
+        # Returns
+            optimization result
+                x_opt (array[1, nx]): x value where fun is at its minimum subject to constraints
+                y_opt (array[1, nx]): fun(x_opt)
+        """
+        ...
+    def minimize(
+        self, func: Callable[[NDArray[np.float64]], NDArray[np.float64]], max_iters: int
+    ) -> OptimResult:
+        """This function finds the minimum of a given function `fun`
+
+        # Parameters
+            max_iters:
+                the iteration budget, number of fun calls is n_doe + q_points * max_iters.
+
+        # Returns
+            optimization result
+                x_opt (array[1, nx]): x value where fun is at its minimum subject to constraints
+                y_opt (array[1, nx]): fun(x_opt)
+        """
+        ...
+    def suggest(
+        self, x_doe: NDArray[np.float64], y_doe: NDArray[np.float64]
+    ) -> NDArray[np.float64]:
+        """This function gives the next best location where to evaluate the function
+        under optimization wrt to previous evaluations.
+        The function returns several point when multi point qEI strategy is used.
+
+        # Parameters
+            x_doe (array[ns, nx]): ns samples where function has been evaluated
+            y_doe (array[ns, 1 + n_cstr]): ns values of objecctive and constraints
+
+
+        # Returns
+            (array[1, nx]): suggested location where to evaluate objective and constraints
+        """
+        ...
+    def get_result_index(self, y_doe: NDArray[np.float64]) -> int:
+        """This function gives the best evaluation index given the outputs
+        of the function (objective wrt constraints) under minimization.
+
+        # Parameters
+            y_doe (array[ns, 1 + n_cstr]): ns values of objective and constraints
+
+        # Returns
+            index in y_doe of the best evaluation
+        """
+        ...
+
+class GpMix:
+    def __init__(
+        self,
+        n_clusters: int = 1,
+        regr_spec: RegressionSpec = RegressionSpec.CONSTANT,
+        corr_spec: CorrelationSpec = CorrelationSpec.SQUAREDEXPONENTIAL,
+        recombination: Recombination = Recombination.HARD,
+        theta_init: list[float] | None = None,
+        theta_bounds: list[list[float]] | None = None,
+        kpls_dim: int | None = None,
+        n_start: int | None = 10,
+        seed: int | None = None,
+    ) -> None: ...
+    def fit(self, xt: NDArray[np.float64], yt: NDArray[np.float64]) -> Gpx: ...
+
+class Gpx:
+    """
+    A trained Gaussian processes mixture.
+    """
+
+    @staticmethod
+    def builder(
+        n_clusters: int = 1,
+        regr_spec: RegressionSpec = RegressionSpec.CONSTANT,
+        corr_spec: CorrelationSpec = CorrelationSpec.SQUAREDEXPONENTIAL,
+        recombination: Recombination = Recombination.SMOOTH,  # XXX: can just ... to be more flexible
```

Comment:
```suggestion
        recombination: Recombination = ...,
```
Can also do this to make the typing more vague so can auto synchronous with the pyo3 wrapper. Not sure which is better.